sw/device/lib/testing/otbn_testutils.c

```diff
@@ -0,0 +1,114 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/otbn_testutils.h"
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/otbn.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+ * Otbn context to be used by the isr handler, it should be assigned with the
+ * current otbn context pointer before enabling the irq.
+ */
+static struct isr_context *otbn_isr_obj;
+
+/**
+ * Provides external IRQ handling for otbn tests.
+ *
+ * This function overrides the default OTTF external ISR.
+ *
+ * It performs the following:
+ * 1. Claims the IRQ fired (finds PLIC IRQ index).
+ * 2. Compute the OTBN peripheral.
+ * 3. Compute the otbn irq.
+ * 4. Clears the IRQ at the peripheral.
+ * 5. Completes the IRQ service at PLIC.
+ */
+void ottf_external_isr(void) {
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_claim(&otbn_isr_obj->plic, kTopEarlgreyPlicTargetIbex0,
+                            (dif_rv_plic_irq_id_t *)&otbn_isr_obj->irq_id));
+
+  otbn_isr_obj->peri = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[otbn_isr_obj->irq_id];
+
+  otbn_isr_obj->irq =
+      (dif_otbn_irq_t)(otbn_isr_obj->irq_id -
+                       (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdOtbnDone);
+
+  CHECK_DIF_OK(dif_otbn_irq_acknowledge(otbn_isr_obj->dif, otbn_isr_obj->irq));
+
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC.
+  // register.
+  CHECK_DIF_OK(dif_rv_plic_irq_complete(
+      &otbn_isr_obj->plic, kTopEarlgreyPlicTargetIbex0, otbn_isr_obj->irq_id));
```

Comment:
Also note, once #11028 and #11029 are merged, there will be a (autogenerated) testutil function you can use to accomplish this. No worries if this PR merges before #11028 and #11029, if that happens I will update this test to make use of the autogenerated ISR testutils. Just wanted to let you know about the autogenerated testutils :)